### ISSUE TITLE
when reading a .pdf file in firefox the entire document is not read

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#14589]
### Steps to reproduce:
open a .pdf document in firefox, I believe a document with at least 15 pages.
start reading the document in a sprint, nvda laptop + a
### Actual behavior:
in my tests for page 3 or 5, nvda will stop reading the document and display the message for all pages:
chart loading page
### Expected behavior:
nvda should read the entire .pdf document like it does in chrome.
It seems that firefox stops loading pages as nvda reads the document, this forces you to stop reading and press ctrl + f5.
it would be interesting for nvda to access new pages automatically as it happens in chrome.
#### NVDA installed/portable/running from source:
portable
#### N
[firefox.txt](https://github.com/nvaccess/nvda/files/10512589/firefox.txt)
VDA version:
nvda.exe, NVDA alpha-27590,180c9f2b

#### Windows version:
10 19045.2251
#### Name and version of other software in use when reproducing the issue:
firefox 109.0
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes
log

**Comment by fernando-jose-silva** [comment]
[firefox.txt](https://github.com/nvaccess/nvda/files/10512667/firefox.txt)

**Comment by jcsteh** [comment]
Firefox doesn't load entire PDF documents because they can potentially be massive (thousands of pages), which would take a very long time to load. However, it should load new pages as you scroll to them. I believe NVDA scrolls during say all, so I'm not sure why this doesn't trigger in this case.

If you down arrow into those pages manually, does the content get loaded?

**Comment by fernando-jose-silva** [comment]
@jcsteh 
thank you very much for the feedback
when reading continues with nvda + a laptop command, and nvda announces
graphic loading page, if you stop reading with ctrl and navigate with the up and down arrows it doesn't make the pages read by nvda, I managed to make nvda read the page by pressing ctrl + f5.

**Comment by jcsteh** [comment]
Could you provide a link to a PDF which reproduces this? I just want to be sure we have consistent reproduction of the issue here.

**Comment by jcsteh** [comment]
I previously believed NVDA would scroll during say all based on [this code](https://github.com/nvaccess/nvda/blob/bb1cde85044dd56fcfb98e6c721920c9b8c20d3d/source/browseMode.py#L1395). However, it definitely doesn't, as demonstrated by this test case:

`data:text/html,<body onscroll="alert('scroll');"><style>hr { height: 100vh; }</style><p>a</p><hr><p>b</p><hr><p>c</p><hr><p>d</p>`

If you use the arrow keys on this page, NVDA scrolls. However, if you use say all, it doesn't.

I now recall why it doesn't work, as I described in https://github.com/nvaccess/nvda/issues/904#issuecomment-316310753.

In short, this issue will be fixed if/when #904 is fixed.
